test_integration/shared.js

```diff
@@ -0,0 +1,89 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+/**
+ * @file `shared` exists so that integration tests can share data among each other.
+ *
+ * One example is the addresses of contracts that were deployed on the test
+ * environment.
+ *
+ * Due to node's caching behavior when loading modules, it always returns the
+ * same object for repeated calls to `require()`.
+ *
+ * It is important that every `require` is written exactly `shared`,
+ * case-sensitive!
+ */
+
+/**
+ * Contains all relevant data on a chain.
+ *
+ * @property {Web3} web3 A web3 object to access the chain.
+ * @property {Object} contractAddresses All addresses in sting format, indexed
+ *     by the contract name, as written in the solidity source file.
+ * @property {Object} contracts Instances of TruffleContracts. Indexed by their
+ *     contract name, e.g. `EIP20Gateway` or `Anchor`. Exception: `BaseToken`
+ *     and `BrandedToken`.
+ * @property {string} deployerAddress The address of the account that deployed
+ *     the contracts in the deployment phase.
+ */
+class Chain {
+    constructor() {
```

Comment:
How integration test will know about owner of organization for the transaction like `anchorStateRoot`?

Reply:
`@property {string} deployerAddress`

Reply:
Because the owner is, by default in the contract, the deployer. Do you want me to copy the deployer address to a key "organization"?

Reply:
Oops, you are right of course. Sorry. I will for now copy the address to an `organization` key. Unless you have a better idea?

Reply:
Sounds good to me.